.github/workflows/ci.yml

```diff
@@ -92,8 +93,13 @@ jobs:
             submodule-
       - uses: actions/checkout@v4
         with:
+          ref: ${{ github.sha }}
```

Comment:
What does this do? I didn't even see the ref get logged, but it looks like it still just checks out the merge commit

Reply:
Currently, when running CI on pushes to master we checkout the repo's master branch as `master` which causes the format check to effectively be skipped. Switching to `github.sha` checks it out as a detached HEAD so format is run on the whole repo instead

Reply:
Could you add a comment then, i forgot this runs on master too